### ISSUE TITLE
Indexer priority for internal searches

### ISSUE THREAD

**Issue by h45rd** [theotherp/nzbhydra2#301]
Hi,

Been doing a lot of manual / internal searches lately.
Wanted to suggest something that might improve usability.

_Respect indexer priorities when grouping posts_
I have some backup indexers with limited API access, in case one of my main indexers misses a post.
While setting appropriate indexer priorities for API searches works as expected. Manual searches will more often than not result in one of the low priority indexers being on the top level of the post grouping, all the while a high priority indexer is in the same group. Hence forcing me to manually expand the group and select the post from the high priority indexer instead of being able to "select all".

While this does not make much of an impact for the occasional manual search for single posts, it can get quite tedious when trying to add a whole series of posts at once.
Especially when there is the need to search more than one indexer at a time, because the main indexer randomly misses e.g. an episode of a series one wants to transparently add from one of the low priority indexers that has it without doing another dedicated search on said indexer.

Long story short: would love it if indexer priority would have an impact on what hydra chooses as top level for duplicate groups and/or which indexer an NZB gets downloaded from when a whole group is selected.

Regards and keep on rocking,
h45rd

**Comment by theotherp** [comment]
The thing is, they're already sorted by indexer score. It just doesn't look like what you expect. Every *duplicate group* is sorted by indexer score. If you don't have any duplicates then you don't see any duplicate groups and therefore don't see the sorting in action. What you're seeing are groups of results with the same name but they're not the same release. If sorted the title groups by indexer score instead of by whatever you're currently searching for then the order between the groups would be inconsistent. 

Say you have two indexers 1 and 2, with the first having score 1 and the second having score 2. Hydra finds four releases:
Release 1 on Indexer 1. 20 days old. 100MB
Release 1 on Indexer 2. 20 days old. 100MB. (exactly the same result as Release 1)
Release 2 on Indexer 1. 10 days old. 210MB
Release 2 on Indexer 1. 30 days old. 200MB (same title, but a different release).

Hydra will recognize that the first two releases are actually the same and show the one from Indexer 1. For release 2 it will recognize that the title is the same but they're actually different releases. When you sort by age descending it will show these releases:
Release 2 on Indexer 1. 10 days old. 210MB
Release 2 on Indexer 2. 30 days old. 200MB (in the same title group)
Release 1 on Indexer 1. 20 days old. 100MB
Release 1 on indexer 2. 20 days old. 100MB is initially hidden as a duplicate.

With everything collapsed that would mean:
Release 2 on Indexer 1. 30 days old. 210MB
Release 1 on Indexer 1. 20 days old. 100MB

If I were to sort the title groups by indexer score and then with collapsed title groups these would be the results shown:
Release 2 on Indexer 1. **10** days old. 200MB
Release 1 on Indexer 1. **20** days old. 100MB

The order of the shown results doesn't fit anymore. I have to sort the results in the title groups the same way the result groups are sorted.

***
Well, I barely understand what I just wrote, so just believe me that it's not easy to accomplish what you want. I literally spent weeks on the sorting and display of the results.

**Comment by h45rd** [comment]
Thanks for taking the time to write this very detailed explanation :)
Now that I (think I) understand the internal sorting logic and handling of duplicate and title groups, it makes perfect sense.

I only checked with two quick sample searches, so please correct me if I am wrong. But it appears that when _explicitly sorting by "Title"_, the releases in the title groups themselves are being sorted by age. Is that correct?

I do get why the default fallback would be to sort by age, given that the titles within a title group are assumed to be identical.
In most cases it makes sense to grab the youngest post.

What would keep us from sorting the title group first by indexer score, then by age in this instance?

**Comment by theotherp** [comment]
I think you're right. I currently don't have enough time for this but I'll keep it in mind.

**Comment by theotherp** [comment]
Was easier than I thought.

**Comment by h45rd** [comment]
Just tested the change. It works like a charm for my use-case and makes adding NZBs in bulk a lot more enjoyable.
You are my personal hero. Thank you very much :)

Have a great weekend,
h45rd

**Comment by theotherp** [comment]
You're welcome. It's possible that other users would prefer the old implementation. We'll see, perhaps I'll make it configurable.

**Comment by h45rd** [comment]
Hehe.. I had to immediately think of this xkcd: https://xkcd.com/1172/
I hope it doesn't break anyone's workflow.

Impact should be rather low, given that one has to explicitly sort by title to override the default.
But I get that the new behavior might be unexpected to some and possibly result in bug reports from those not closely following the release notes.

You're right. Making the behavior configurable would probably be the best solution in the long run.
